### PR TITLE
Remove Server's dependency on Controller classes

### PR DESCRIPTION
**Pull request by ThanGerlek**
Introduces an EndpointProvider interface which Server now uses for all its endpoint calls. Also extracts startup logic (getOptions, initializeDaos, etc) out of the Server class and into a Main class.

> [!WARNING]
> This moves the location of the main() method (from `java/server/Server.java` to `java/Main.java`). Existing run configurations will likely break and will need to be updated.

### CODE REVIEW

**Comment by ThanGerlek** [comment]
@19mdavenport Since this moves the main() method into a different file, how will this affect the existing deployment system? Does anything else need to happen?

The good news is that it's now in a dedicated Main class, so it will probably never need to move again.

**Review by webecke — APPROVED** [review]
LGTM!

Seems like a clean solution to make things more independent of each other and more flexible. It does look like adding new endpoints now requires a little more work, but its marginal so I'm not too worried about it. Great work!

File: src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+
+public class Main {
+    private static Logger LOGGER = LoggerFactory.getLogger(Main.class);
+
+    private static EndpointProvider endpointProvider = new EndpointProviderImpl();
+
+    public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (19mdavenport) [reply src/main/java/Main.java:23]: I think so, you could try testing this by running `mvn exec:java` in the base directory and seeing if the server starts normally.

[the remark this answers is not in the document]
Comment (19mdavenport) [reply src/main/java/Main.java:23]: Also, did you commit/push that update?

**Review by 19mdavenport — APPROVED** [review]

File: src/main/java/Main.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThanGerlek) [reply src/main/java/Main.java:23]: Yes, but in the wrong branch. RIP.

**Comment by frozenfrank** [comment]
`git rebase —onto NEW_BRANCH HEAD^`

**Comment by frozenfrank** [comment]
`git checkout CORRECT_BRANCH; git cherry-pick -x CORRECT_COMMIT`

**Comment by ThanGerlek** [comment]
@19mdavenport Apparently `mvn exec:java` doesn't work for me (I get [this error message](https://stackoverflow.com/questions/19850956/maven-java-the-parameters-mainclass-for-goal-org-codehaus-mojoexec-maven-p) (although the listed solutions don't fix it for me), even on the main branch), so someone other than me might need to test that. Obviously it does pass our CI/CD tests but I assume they have a different execution path

**Pull request merged**